open_spiel/python/games/liars_poker.py

```diff
@@ -0,0 +1,379 @@
+    information=pyspiel.GameType.Information.IMPERFECT_INFORMATION,
+    utility=pyspiel.GameType.Utility.ZERO_SUM,
+    reward_model=pyspiel.GameType.RewardModel.TERMINAL,
+    max_num_players=_MAX_NUM_PLAYERS,
+    min_num_players=_MIN_NUM_PLAYERS,
+    provides_information_state_string=True,
+    provides_information_state_tensor=True,
+    provides_observation_string=False,
+    provides_observation_tensor=True,
+    parameter_specification={
+      "players": _MIN_NUM_PLAYERS,
+      "hand_length": _HAND_LENGTH,
+      "num_digits": _NUM_DIGITS
+    })
+_GAME_INFO = pyspiel.GameInfo(
+    # Num actions = total number of cards * number of digits + action enum
+    num_distinct_actions=_HAND_LENGTH * _NUM_DIGITS * _MIN_NUM_PLAYERS + len(Action),
+    max_chance_outcomes=_HAND_LENGTH * _NUM_DIGITS,
+    num_players=_MIN_NUM_PLAYERS,
+    min_utility=-(_MIN_NUM_PLAYERS - 1), # Reward from being challenged and losing.
+    max_utility=_MIN_NUM_PLAYERS - 1, # Reward for being challenged and winning.
+    utility_sum=0.0,
+    # Number of possible rounds: hand_length * num_digits * num_players
+    # Total moves per round: num_players for non-rebid, num_players-1 for rebid
+    # Max game length: number of possible rounds * total moves per round
+    max_game_length=_HAND_LENGTH * _NUM_DIGITS * _MIN_NUM_PLAYERS**2)
+
+class LiarsPoker(pyspiel.Game):
+  """A Python version of Liar's poker."""
+
+  def __init__(self, params=None):
+    super().__init__(_GAME_TYPE, _GAME_INFO, params or dict())
+    game_parameters = self.get_parameters()
+    self.hand_length = game_parameters.get("hand_length", _HAND_LENGTH)
+    self.num_digits = game_parameters.get("num_digits", _NUM_DIGITS)
+    self.deck = [_FULL_DECK[i] for i in range(self.num_digits)]
+
+  def new_initial_state(self):
+    """Returns a state corresponding to the start of a game."""
+    return LiarsPokerState(self)
+
+  def make_py_observer(self, iig_obs_type=None, params=None):
+    """Returns an object used for observing game state."""
+    return LiarsPokerObserver(
+      iig_obs_type or pyspiel.IIGObservationType(perfect_recall=False),
+      self.num_players(),
+      self.hand_length,
+      self.num_digits,
+      params)
+
+
+class LiarsPokerState(pyspiel.State):
+  """A python version of the Liars Poker state."""
+
+  def __init__(self, game):
+    """Constructor; should only be called by Game.new_initial_state."""
+    super().__init__(game)
+    # Game attributes
+    self._num_players = game.num_players()
+    self._hand_length = game.hand_length
+    self._num_digits = game.num_digits
+    self._deck = game.deck
+    self.hands = [[] for _ in range(self._num_players)]
+
+    # Action dynamics
+    total_possible_bets = game.hand_length * game.num_digits * self._num_players
+    self.bid_history = np.zeros((total_possible_bets, self._num_players))
+    self.challenge_history = np.zeros((total_possible_bets, self._num_players))
+    self._current_player = 0
```

Comment:
Maybe add short comment that self._current_player is only the valid current_player when cards have been dealt (otherwise it's chance)